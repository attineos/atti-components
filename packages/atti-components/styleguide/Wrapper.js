import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
