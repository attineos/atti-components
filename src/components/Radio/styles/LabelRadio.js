import styled from 'styled-components'
import RadioButton from './RadioButton'
import HiddenInputRadio from './HiddenInputRadio'
import Label from '../../Label'

const LabelRadio = styled(Label)`
  display: flex;
  align-items: center;
  cursor: pointer;
  position: relative;

  // Modifications of the color background of the radio :
  &:hover ${HiddenInputRadio} + ${RadioButton} {
    &:after {
      display: block;
      background-color: ${({ theme }) => theme.components.radio.radioButton.colors.backgroundHover};
    }
  }

  ${HiddenInputRadio}:checked + ${RadioButton} {
    &:after {
      display: block;
    }
  }
`

export default LabelRadio
