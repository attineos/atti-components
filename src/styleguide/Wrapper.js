import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
