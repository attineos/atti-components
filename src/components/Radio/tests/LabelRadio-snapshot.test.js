/* This file was generated automatically, please do not write in it. */
import React from 'react';
import LabelRadio from '../styles/LabelRadio';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('LabelRadio renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <LabelRadio ></LabelRadio>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    