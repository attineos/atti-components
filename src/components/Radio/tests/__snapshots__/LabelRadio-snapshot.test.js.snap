// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelRadio renders correctly 1`] = `
<label
  className="LabelRadio-wot4p6-0 jilFc"
/>
`;
